src/main/scala/org/constellation/p2p/PeerAPI.scala

```diff
@@ -183,30 +181,24 @@ class PeerAPI(override val ipManager: IPManager, nodeActor: ActorRef)(
         pathPrefix("finished") {
           path("checkpoint") {
 
+            implicit val ioContextShift: ContextShift[IO] =
+              IO.contextShift(dao.finishedExecutionContext)
+
             extractClientIP { ip =>
-              /*
-            ip.toOption.foreach { inet =>
-              val hp = HostPort(inet.getHostAddress, 9001) // TODO: Change this to non-hardcoded port and send a response telling other node to re-register
-              PeerManager.attemptRegisterPeer(hp)
-            }
-               */
 
               entity(as[FinishedCheckpoint]) { fc =>
                 optionalHeaderValueByName("ReplyTo") { replyToOpt =>
-                  val maybeData = getHostAndPortFromRemoteAddress(ip)
-                  val knownHost = maybeData.exists(
-                    i => dao.peerInfo.unsafeRunSync().exists(_._2.client.hostName == i.canonicalHostName)
-                  )
+
                   dao.metrics.incrementMetric("peerApiRXFinishedCheckpoint")
-                  EdgeProcessor.handleFinishedCheckpoint(fc).map { result =>
+
+                  (IO.shift *> dao.checkpointService.accept(fc)).unsafeToFuture().onComplete { result =>
```

Comment:
In my opinion. it's better to create a context shift not as an implicit value and write here `ioContextShift.shift *> ...`
This way you're able to limit the number of implicits